--- a/src/graphic/text/bidi.h	2018-09-29 19:19:14.884233816 +0200
+++ b/src/graphic/text/bidi.h	2018-09-29 19:22:30.742229163 +0200
@@ -40,7 +40,11 @@
 // True if a string contains a character from the script's code blocks
 bool has_script_character(const char* input, UI::FontSets::Selector script);
 UI::FontSet const* find_fontset(const char* word, const UI::FontSets& fontsets);
+#if U_ICU_VERSION_MAJOR_NUM > 60
+std::string icustring2string(const icu::UnicodeString& convertme);
+#else
 std::string icustring2string(const UnicodeString& convertme);
+#endif // >=icu-61.1
 std::string icuchar2string(const UChar& convertme);
 bool cannot_start_line(const UChar& c);
 bool cannot_end_line(const UChar& c);
