--- cs-config-2.0	2012-04-20 15:46:07.652756874 +0200
+++ cs-config-2.0	2012-04-20 16:00:40.872802423 +0200
@@ -91,7 +91,7 @@
 then
   includedir="/usr/include/crystalspace-2.0"
 fi
-syslibs=" -lc -lm -ldl -lnsl -L/usr/local/lib "
+syslibs=" -lc -lm -ldl -lnsl "
 common_cflags="  "
 common_cxxflags="   "
 staticdeps="crystalspace_staticplugins.jam"
@@ -106,7 +106,7 @@
 EXE=
 DLL=.so
 LIBS.EXE.PLATFORM=-ldl -lm -lnsl
-LFLAGS.GENERAL=$(LFLAGS.L)/usr/local/lib -lpthread
+LFLAGS.GENERAL=$(LFLAGS.L)/usr/lib -lpthread
 LFLAGS.DLL=-shared -Wl,-soname -Wl,$@
 
 __EOF__
@@ -156,14 +156,12 @@
     esac
 }
 
-liblist=" crystalspace crystalspace_opengl
-"
-liblist_opt=" crystalspace_staticplugins
-"
+liblist=" crystalspace crystalspace_opengl"
+liblist_opt=" crystalspace_staticplugins"
 
 libcrystalspace=crystalspace-2.0
- libcrystalspace_opengl=crystalspace_opengl-2.0
- libcrystalspace_staticplugins=crystalspace_staticplugins-2.0
+libcrystalspace_opengl=crystalspace_opengl-2.0
+libcrystalspace_staticplugins=crystalspace_staticplugins-2.0
 
 
 
@@ -223,6 +221,14 @@
     fi
 done
 
+# Patch: On Gentoo they are always there:
+if test "x$lflags" = "x"
+then
+  libdir="/usr/lib"
+  lflags="-L$libdir"
+fi
+
+
 # Look for build tools such as relaytool
 tools_sentinel=relaytool
 if test -x "$prefix/bin/$tools_sentinel" ; then
