--- configure.ac	2012-04-27 21:27:38.556322395 +0200
+++ configure.ac	2012-04-27 21:36:42.666350846 +0200
@@ -2773,7 +2773,19 @@
 
 #------------------------------------------------------------------------------
 # Put FLAGS together
+# To remove cruft we first move everything now set in the ENV vars to shell
+# vars and then move them back.
 #------------------------------------------------------------------------------
+CS_APPEND_TRIMMED([cs_cflags], [$CFLAGS])
+CS_APPEND_TRIMMED([cs_cxxflags], [$CXXFLAGS])
+CS_APPEND_TRIMMED([cs_cppflags], [$CPPFLAGS])
+CS_APPEND_TRIMMED([cs_ldflags], [$LDFLAGS])
+CS_ERASE_VAR(CFLAGS)
+CS_ERASE_VAR(CPPFLAGS)
+CS_ERASE_VAR(CXXFLAGS)
+CS_ERASE_VAR(LDFLAGS)
+
+# Set ENV FLAGS
 CS_APPEND_TRIMMED([CFLAGS], [$cs_cflags])
 CS_APPEND_TRIMMED([CPPFLAGS], [$cs_cppflags])
 CS_APPEND_TRIMMED([CXXFLAGS], [$cs_cxxflags])
@@ -2783,6 +2795,30 @@
 CS_EMIT_BUILD_PROPERTY([COMPILER.C++FLAGS], [$CXXFLAGS], [+])
 CS_EMIT_BUILD_PROPERTY([COMPILER.LFLAGS], [$LDFLAGS], [+])
 
+# Set debug flags
+CS_EMIT_BUILD_PROPERTY([COMPILER.CFLAGS.debug], [$cs_cflags_debug], [+])
+CS_EMIT_BUILD_PROPERTY([COMPILER.CPPFLAGS.debug], [$cs_cppflags_debug], [+])
+CS_EMIT_BUILD_PROPERTY([COMPILER.C++FLAGS.debug], [$cs_cxxflags_debug], [+])
+CS_EMIT_BUILD_PROPERTY([COMPILER.LFLAGS.debug], [$cs_ldflags_debug], [+])
+
+# Set optimize flags
+CS_EMIT_BUILD_PROPERTY([COMPILER.CFLAGS.optimize], [$cs_cflags_optimize], [+])
+CS_EMIT_BUILD_PROPERTY([COMPILER.CPPFLAGS.optimize], [$cs_cppflags_optimize], [+])
+CS_EMIT_BUILD_PROPERTY([COMPILER.C++FLAGS.optimize], [$cs_cxxflags_optimize], [+])
+CS_EMIT_BUILD_PROPERTY([COMPILER.LFLAGS.optimize], [$cs_ldflags_optimize], [+])
+
+# Set profile flags
+CS_EMIT_BUILD_PROPERTY([COMPILER.CFLAGS.profile], [$cs_cflags_profile], [+])
+CS_EMIT_BUILD_PROPERTY([COMPILER.CPPFLAGS.profile], [$cs_cppflags_profile], [+])
+CS_EMIT_BUILD_PROPERTY([COMPILER.C++FLAGS.profile], [$cs_cxxflags_profile], [+])
+CS_EMIT_BUILD_PROPERTY([COMPILER.LFLAGS.profile], [$cs_ldflags_profile], [+])
+
+# Set custom flags
+CS_EMIT_BUILD_PROPERTY([COMPILER.CFLAGS.custom], [$cs_cflags_custom], [+])
+CS_EMIT_BUILD_PROPERTY([COMPILER.CPPFLAGS.custom], [$cs_cppflags_custom], [+])
+CS_EMIT_BUILD_PROPERTY([COMPILER.C++FLAGS.custom], [$cs_cxxflags_custom], [+])
+CS_EMIT_BUILD_PROPERTY([COMPILER.LFLAGS.custom], [$cs_ldflags_custom], [+])
+
 
 #------------------------------------------------------------------------------
 # Boilerplate: csconfig.h
