--- plugins/physics/odedynam/odedynam.cpp	2012-01-14 04:39:50.000000000 +0100
+++ plugins/physics/odedynam/odedynam.cpp	2012-07-13 09:38:15.254617788 +0200
@@ -798,20 +798,13 @@
   while (total_elapsed > stepsize)
   {
     total_elapsed -= stepsize;
-    if (!stepfast)
+    if (!quickstep)
     {
-      if (!quickstep)
-      {
-        dWorldStep (worldID, stepsize);
-      }
-      else
-      {
-        dWorldQuickStep (worldID, stepsize);
-      }
+      dWorldStep (worldID, stepsize);
     }
     else
     {
-      dWorldStepFast1 (worldID, stepsize, sfiter);
+      dWorldQuickStep (worldID, stepsize);
     }
     for (size_t i = 0; i < bodies.GetSize (); i ++)
     {
