--- configure.ac	2012-04-24 08:32:35.536552189 +0200
+++ configure.ac	2012-04-24 08:51:56.386613019 +0200
@@ -874,23 +874,34 @@
 # attempts to run a test program. Without augmenting LD_LIBRARY_PATH, the test
 # program would fail to run, citing inability to locate libstdc++.so.
 #------------------------------------------------------------------------------
-AS_IF([test -d /usr/local/include],
-    [CS_CHECK_BUILD_FLAGS([if -I/usr/local/include is usable],
-	[cs_cv_prog_cpp_local_include],
-	[CS_CREATE_TUPLE([-I/usr/local/include])], [C++], [], [],
-	[$cs_cv_prog_cxx_enable_warnings $cs_cv_prog_cxx_enable_errors])],
+# Update, 2012-04-24, Sven Eden:
+# If the user adds a specific includedir or libdir or those directories are
+# changed by the --prefix option, it is a nuisance to have /usr/local/foo added
+# all over the place. Thus do not do anything unless the configure options say
+# so. Furthermore the user is allowed to use any --prefix they like, so do not
+# Hardcode the default path here.
+#------------------------------------------------------------------------------
+AS_IF([test "x$includedir" != "x/usr/include"],
+    AS_IF([test -d $includedir],
+        [CS_CHECK_BUILD_FLAGS([if -I$includedir is usable],
+            [cs_cv_prog_cpp_local_include],
+            [CS_CREATE_TUPLE([-I$includedir])], [C++], [], [],
+            [$cs_cv_prog_cxx_enable_warnings $cs_cv_prog_cxx_enable_errors])],
+        [cs_cv_prog_cpp_local_include='']),
     [cs_cv_prog_cpp_local_include=''])
 
+
 AS_IF([test -n "$cs_cv_prog_cpp_local_include"],
     [CS_EMIT_BUILD_PROPERTY([COMPILER.CFLAGS],
-	[$cs_cv_prog_cpp_local_include], [+])
+        [$cs_cv_prog_cpp_local_include], [+])
     CPPFLAGS="$CPPFLAGS $cs_cv_prog_cpp_local_include"])
 
-AS_IF([test -d /usr/local/lib],
-    [CS_EMIT_BUILD_PROPERTY([COMPILER.LFLAGS], [-L/usr/local/lib], [+])
-    LDFLAGS="$LDFLAGS -L/usr/local/lib"
-    LD_LIBRARY_PATH="$LD_LIBRARY_PATH$PATH_SEPARATOR/usr/local/lib"
-    export LD_LIBRARY_PATH])
+AS_IF([test -d $libdir],
+    [CS_EMIT_BUILD_PROPERTY([COMPILER.LFLAGS], [-L$libdir], [+])
+    LDFLAGS="$LDFLAGS -L$libdir"
+    AS_IF([test "x$libdir" != "x/usr/lib" -a "x$libdir" != "x/usr/lib32" -a "x$libdir" != "x/usr/lib64"],
+        [LD_LIBRARY_PATH="$LD_LIBRARY_PATH$PATH_SEPARATOR$libdir"
+        export LD_LIBRARY_PATH])])
 
 
 
