--- scripts/perl5/cspace.pm	2012-01-14 04:39:45.000000000 +0100
+++ scripts/perl5/cspace.pm	2012-07-13 09:58:28.784680943 +0200
@@ -7615,10 +7615,6 @@
 *GlobalERP = *cspacec::iODEDynamicState_GlobalERP;
 *SetGlobalCFM = *cspacec::iODEDynamicState_SetGlobalCFM;
 *GlobalCFM = *cspacec::iODEDynamicState_GlobalCFM;
-*EnableStepFast = *cspacec::iODEDynamicState_EnableStepFast;
-*StepFastEnabled = *cspacec::iODEDynamicState_StepFastEnabled;
-*SetStepFastIterations = *cspacec::iODEDynamicState_SetStepFastIterations;
-*StepFastIterations = *cspacec::iODEDynamicState_StepFastIterations;
 *EnableQuickStep = *cspacec::iODEDynamicState_EnableQuickStep;
 *QuickStepEnabled = *cspacec::iODEDynamicState_QuickStepEnabled;
 *SetQuickStepIterations = *cspacec::iODEDynamicState_SetQuickStepIterations;
@@ -7671,10 +7667,6 @@
 *ERP = *cspacec::iODEDynamicSystemState_ERP;
 *SetCFM = *cspacec::iODEDynamicSystemState_SetCFM;
 *CFM = *cspacec::iODEDynamicSystemState_CFM;
-*EnableStepFast = *cspacec::iODEDynamicSystemState_EnableStepFast;
-*StepFastEnabled = *cspacec::iODEDynamicSystemState_StepFastEnabled;
-*SetStepFastIterations = *cspacec::iODEDynamicSystemState_SetStepFastIterations;
-*StepFastIterations = *cspacec::iODEDynamicSystemState_StepFastIterations;
 *EnableQuickStep = *cspacec::iODEDynamicSystemState_EnableQuickStep;
 *QuickStepEnabled = *cspacec::iODEDynamicSystemState_QuickStepEnabled;
 *SetQuickStepIterations = *cspacec::iODEDynamicSystemState_SetQuickStepIterations;
