--- include/csutil/redblacktree.h.ori	2012-07-18 17:23:56.847708962 +0200
+++ include/csutil/redblacktree.h	2012-07-18 17:26:29.221050599 +0200
@@ -1155,7 +1155,7 @@
   T* Put (const K& key, const T &value)
   {
     csRedBlackTreePayload<K, T>* payload = (csRedBlackTreePayload<K, T>*)
-      Insert (csRedBlackTreePayload<K, T>(key, value));
+      this->Insert (csRedBlackTreePayload<K, T>(key, value));
     return (payload != 0) ? &payload->GetValue() :  0;
   }
   /**
@@ -1194,7 +1194,7 @@
    */
   const T& Get (const K& key, const T& fallback) const
   {
-    const csRedBlackTreePayload<K, T>* payload = Find (key);
+    const csRedBlackTreePayload<K, T>* payload = this->Find (key);
     if (payload == 0) return fallback;
     return payload->GetValue();
   }
