--- include/ivaria/ode.h	2012-01-14 04:41:01.000000000 +0100
+++ include/ivaria/ode.h	2012-07-13 09:57:20.284677366 +0200
@@ -56,12 +56,6 @@
   virtual void SetGlobalCFM (float cfm) = 0;
   virtual float GlobalCFM () = 0;
 
-  /// Enables the experimental StepFast code in ode
-  virtual void EnableStepFast (bool enable) = 0;
-  virtual bool StepFastEnabled () = 0;
-  virtual void SetStepFastIterations (int iter) = 0;
-  virtual int StepFastIterations () = 0;
-
   virtual void EnableQuickStep (bool enable) = 0;
   virtual bool QuickStepEnabled () = 0;
   virtual void SetQuickStepIterations (int iter) = 0;
@@ -151,16 +145,6 @@
   virtual void SetCFM (float cfm) = 0;
   virtual float CFM () = 0;
 
-  /**
-   * Enables the experimental StepFast code in ode
-   * Setting this in ODEDynamicState sets it here
-   * Only modify it if you want a specific system to behave differently
-   */
-  virtual void EnableStepFast (bool enable) = 0;
-  virtual bool StepFastEnabled () = 0;
-  virtual void SetStepFastIterations (int iter) = 0;
-  virtual int StepFastIterations () = 0;
-
   virtual void EnableQuickStep (bool enable) = 0;
   virtual bool QuickStepEnabled () = 0;
   virtual void SetQuickStepIterations (int iter) = 0;
