--- data/sky/Jamfile	2012-04-23 21:22:04.833271060 +0200
+++ data/sky/Jamfile	2012-04-23 21:23:55.199944160 +0200
@@ -1,4 +1,7 @@
 SubDir TOP data sky ;
 
-InstallData [ Wildcard *.layout *.xml ] : data sky ;
-FileListEntriesData [ Wildcard *.layout *.xml ] : sky : data-demo ;
+{
+local files = library [ Wildcard *.png ] ;
+InstallData $(files) : data sky ;
+FileListEntriesData $(files) : sky : data-demo ;
+}
