--- mk/jam/variant.jam	2012-04-27 20:49:21.256202254 +0200
+++ mk/jam/variant.jam	2012-04-27 20:50:56.272873856 +0200
@@ -23,9 +23,9 @@
   VARIANT = $(MODE) ;
 }
 
-if $(VARIANT) != "debug" && $(VARIANT) != "optimize" && $(VARIANT) != "profile"
+if $(VARIANT) != "debug" && $(VARIANT) != "optimize" && $(VARIANT) != "profile" && $(VARIANT) != "custom"
 {
-  exit "Invalid modus set, please set VARIANT to debug, profile or optimize" ;
+  exit "Invalid modus set, please set VARIANT to debug, profile, optimize or custom" ;
 }
 
 # Set modus related flags
