--- data/Jamfile	2012-04-23 17:12:32.953141311 +0200
+++ data/Jamfile	2012-04-23 17:33:29.893158458 +0200
@@ -24,25 +24,38 @@
 FileListEntriesData $(RUNTIME_DATA) : : data-runtime ;
 
 maps =
+  bias
+  colladatest
+  island
   isomap
+  lodtest
+  oceantest
   parallaxtest
-  particles
   partsys
+  projectortest
+  r3dsmtest
   r3dtest
+  space
   stenciltest
   terrain
-  terrainf ;
+  terraina
+  terrainf
+  terraini ;
 for map in $(maps)
 {
   local files =
-    [ Wildcard $(map) : *.lib *.jpg *.png *.map *.tga *.dds *.spr world ] ;
+    [ Wildcard $(map) : *.cslib *.lib *.jpg *.png *.map *.tga *.dds *.spr *.zip world ] ;
   local lmfiles =
     [ Wildcard $(map) lightmaps : *.png ] ;
+  local mffiles =
+    [ Wildcard $(map) factories : *.cslib *.meshfact ] ;
   InstallData $(files) : data maps $(map) ;
   InstallData $(lmfiles) : data maps $(map) lightmaps ;
+  InstallData $(mffiles) : data maps $(map) factories ;
 
   FileListEntriesData $(files) : $(map) : data-demo-maps ;
   FileListEntriesData $(lmfiles) : $(map) lightmaps : data-demo-maps ;
+  FileListEntriesData $(mffiles) : $(map) factories : data-demo-maps ;
 }
 
 SubInclude TOP data castle ;
@@ -55,12 +68,13 @@
 SubInclude TOP data flarge ;
 SubInclude TOP data frankie ;
 SubInclude TOP data hairtest ;
-SubInclude TOP data sky ;
 SubInclude TOP data krystal ;
 SubInclude TOP data posteffects ;
 SubInclude TOP data renderlayers ;
 SubInclude TOP data shader ;
 SubInclude TOP data shader-old ;
 SubInclude TOP data shader-snippets ;
+SubInclude TOP data sky ;
+SubInclude TOP data smoketest ;
 SubInclude TOP data varia ;
 SubInclude TOP data viewmesh ;
