--- include/csutil/blockallocator.h.ori	2012-07-18 16:20:00.460841926 +0200
+++ include/csutil/blockallocator.h	2012-07-18 17:02:35.940975474 +0200
@@ -193,7 +193,7 @@
   ~csBlockAllocator()
   {
     ObjectDispose dispose (*this, false);
-    DisposeAll (dispose);
+    this->DisposeAll (dispose);
   }
 
   /**
@@ -204,7 +204,7 @@
   void Empty ()
   {
     ObjectDispose dispose (*this, true);
-    FreeAll (dispose);
+    this->FreeAll (dispose);
   }
 
   /**
@@ -215,7 +215,7 @@
   void DeleteAll ()
   {
     ObjectDispose dispose (*this, true);
-    DisposeAll (dispose);
+    this->DisposeAll (dispose);
   }
 
   /**
