--- scripts/python/frozen/cspace/ivaria.py	2012-01-14 04:39:47.000000000 +0100
+++ scripts/python/frozen/cspace/ivaria.py	2012-07-13 10:11:57.391389919 +0200
@@ -774,10 +774,6 @@
     def GlobalERP(*args): return _ivaria.iODEDynamicState_GlobalERP(*args)
     def SetGlobalCFM(*args): return _ivaria.iODEDynamicState_SetGlobalCFM(*args)
     def GlobalCFM(*args): return _ivaria.iODEDynamicState_GlobalCFM(*args)
-    def EnableStepFast(*args): return _ivaria.iODEDynamicState_EnableStepFast(*args)
-    def StepFastEnabled(*args): return _ivaria.iODEDynamicState_StepFastEnabled(*args)
-    def SetStepFastIterations(*args): return _ivaria.iODEDynamicState_SetStepFastIterations(*args)
-    def StepFastIterations(*args): return _ivaria.iODEDynamicState_StepFastIterations(*args)
     def EnableQuickStep(*args): return _ivaria.iODEDynamicState_EnableQuickStep(*args)
     def QuickStepEnabled(*args): return _ivaria.iODEDynamicState_QuickStepEnabled(*args)
     def SetQuickStepIterations(*args): return _ivaria.iODEDynamicState_SetQuickStepIterations(*args)
@@ -809,10 +805,6 @@
     def ERP(*args): return _ivaria.iODEDynamicSystemState_ERP(*args)
     def SetCFM(*args): return _ivaria.iODEDynamicSystemState_SetCFM(*args)
     def CFM(*args): return _ivaria.iODEDynamicSystemState_CFM(*args)
-    def EnableStepFast(*args): return _ivaria.iODEDynamicSystemState_EnableStepFast(*args)
-    def StepFastEnabled(*args): return _ivaria.iODEDynamicSystemState_StepFastEnabled(*args)
-    def SetStepFastIterations(*args): return _ivaria.iODEDynamicSystemState_SetStepFastIterations(*args)
-    def StepFastIterations(*args): return _ivaria.iODEDynamicSystemState_StepFastIterations(*args)
     def EnableQuickStep(*args): return _ivaria.iODEDynamicSystemState_EnableQuickStep(*args)
     def QuickStepEnabled(*args): return _ivaria.iODEDynamicSystemState_QuickStepEnabled(*args)
     def SetQuickStepIterations(*args): return _ivaria.iODEDynamicSystemState_SetQuickStepIterations(*args)
