--- mk/autoconf/trim.m4	2012-04-26 07:21:42.859168666 +0200
+++ mk/autoconf/trim.m4	2012-04-26 07:39:40.795891383 +0200
@@ -25,3 +25,53 @@
 #	runs of multiple spaces to a single space.
 #------------------------------------------------------------------------------
 AC_DEFUN([CS_TRIM], [`echo x$1 | sed 's/^x//;s/   */ /g;s/^ //;s/ $//'`])
+
+#------------------------------------------------------------------------------
+# CS_TRIM_OPT(shell var)
+# Remove leading dashes
+#------------------------------------------------------------------------------
+AC_DEFUN([CS_TRIM_OPT], [`echo x$1 | sed 's/^x//;s/^-*//g'`])
+
+#------------------------------------------------------------------------------
+# CS_APPEND_TRIMMED(named shell var, STRING)
+# Go through all entries in STRING and append those that are not present in the
+# given named shell variable.
+#------------------------------------------------------------------------------
+AC_DEFUN([CS_APPEND_TRIMMED],
+  [AS_IF([test -n "$2"],
+    [append_text="$$1"
+    append_values="CS_TRIM([$2])"
+    for value in $append_values ; do
+      append_found=''
+      AS_IF([test -n "$append_text"],
+      [chkVal="CS_TRIM_OPT([$value])"
+      append_found="`echo -n "$append_text" | grep -o "$chkVal"`"])
+      AS_IF([test -z "$append_found"],
+        [$1="${$1} ${value}"
+        append_text="$$1"])
+    done])])
+
+#------------------------------------------------------------------------------
+# CS_PREPEND_TRIMMED(named shell var, STRING)
+# Go through all entries in STRING and prepend those that are not present in
+# the given named shell variable.
+#------------------------------------------------------------------------------
+AC_DEFUN([CS_PREPEND_TRIMMED],
+  [AS_IF([test -n "$2"],
+    [prepend_text="$$1"
+    prepend_values="CS_TRIM([$2])"
+    for value in $prepend_values ; do
+      prepend_found=''
+      AS_IF([test -n "$prepend_text"],
+      [chkVal="CS_TRIM_OPT([$value])"
+      prepend_found="`echo -n "$prepend_text" | grep -o "$chkVal"`"])
+      AS_IF([test -z "$prepend_found"],
+        [$1="${value} ${$1}"
+        prepend_text="$$1"])
+    done])])
+
+#------------------------------------------------------------------------------
+# CS_ERASE_VAR(named shell var)
+# Simply set the named shell var to an empty string
+#------------------------------------------------------------------------------
+AC_DEFUN([CS_ERASE_VAR], [AS_IF([test -n "$1"], [$1=''])])
