--- data/flarge/Jamfile	2012-04-23 17:38:46.486508351 +0200
+++ data/flarge/Jamfile	2012-04-23 17:38:46.489841685 +0200
@@ -1,10 +1,9 @@
 SubDir TOP data flarge ;
 
-InstallData world [ Wildcard *.png *.dds ] : data maps flarge ;
-InstallData [ Wildcard factories : thing* ] : data maps flarge factories ;
+InstallData world [ Wildcard *.png *.dds *.cslib ] : data maps flarge ;
+InstallData [ Wildcard factories : *.meshfact ] : data maps flarge factories ;
 InstallData [ Wildcard lightmaps : *.png ] : data maps flarge lightmaps ;
 
-FileListEntriesData world [ Wildcard *.png *.dds ] : flarge : data-demo-maps ;
-FileListEntriesData [ Wildcard factories : thing* ] : flarge factories : data-demo-maps ;
+FileListEntriesData world [ Wildcard *.png *.dds *.cslib ] : flarge : data-demo-maps ;
+FileListEntriesData [ Wildcard factories : *.meshfact ] : flarge factories : data-demo-maps ;
 FileListEntriesData [ Wildcard lightmaps : *.png ] : flarge lightmaps : data-demo-maps ;
-
