--- a/CMakeLists.txt	2021-06-17 12:01:11.938991842 +0200
+++ b/CMakeLists.txt	2021-06-17 12:01:39.312993419 +0200
@@ -436,7 +436,7 @@
 if(UNIX AND NOT APPLE)
   install(FILES docs/melt.1 DESTINATION ${CMAKE_INSTALL_MANDIR}/man1 RENAME melt-${MLT_VERSION_MAJOR}.1)
   install(CODE "execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink melt-${MLT_VERSION_MAJOR}.1 melt.1 \
-                                WORKING_DIRECTORY ${CMAKE_INSTALL_FULL_MANDIR}/man1)"
+                                WORKING_DIRECTORY \$ENV\{DESTDIR\}${CMAKE_INSTALL_FULL_MANDIR}/man1)"
   )
 endif()
 
