--- ./MEGASync/platform/platform.pri.ori	2017-12-29 08:14:10.123439659 +0100
+++ ./MEGASync/platform/platform.pri	2017-12-29 09:10:06.323377579 +0100
@@ -26,12 +26,12 @@
 	
 	distro.target = $$PWD/linux/data/distro
     distro.path = $$DATADIR/doc/megasync
-    system(command -v lsb_release): distro.commands = lsb_release -ds > $$distro.target
+    distro.commands = cat /etc/gentoo-release > $$distro.target
     distro.files = $$distro.target
     
     version.target = $$PWD/linux/data/version
 	version.path = $$DATADIR/doc/megasync
-    system(command -v lsb_release): version.commands = lsb_release -rs > $$version.target
+    version.commands = cut -d \' \' -f 5 /etc/gentoo-release > $$version.target
 	version.files = $$version.target
 
 	INSTALLS += distro version
