--- a/MEGASync/MEGASync.pro	2017-12-28 09:21:38.917418127 +0100
+++ b/MEGASync/MEGASync.pro	2017-12-28 09:22:28.967419630 +0100
@@ -28,11 +28,11 @@
     TARGET = megasync
 
 #    Uncomment the following if "make install" doesn't copy megasync in /usr/bin directory
-#    isEmpty(PREFIX) {
-#        PREFIX = /usr
-#    }
-#    target.path = $$PREFIX/bin
-#    INSTALLS += target
+    isEmpty(PREFIX) {
+        PREFIX = /usr
+    }
+    target.path = $$PREFIX/bin
+    INSTALLS += target
 }
 else {
     TARGET = MEGAsync
