--- a/MEGASync/mega/src/posix/fs.cpp	2018-10-01 12:57:20.853946247 +0200
+++ b/MEGASync/mega/src/posix/fs.cpp	2018-10-01 13:16:34.393906698 +0200
@@ -1739,7 +1739,8 @@
     // FIXME: statfs() does not really do what we want.
     if (statfs(localbasepath.c_str(), &statfsbuf)) return 0;
 
-    return *(fsfp_t*)&statfsbuf.f_fsid + 1;
+    return (fsfp_t)( (uint64_t)statfsbuf.f_fsid.__val[0]
+                   | (uint64_t)statfsbuf.f_fsid.__val[1] << 32) + 1;
 }
 
 FileAccess* PosixFileSystemAccess::newfileaccess()
