--- CMakeLists.txt~	2012-11-14 09:53:20.000000000 +0100
+++ CMakeLists.txt	2012-11-14 10:07:07.823249858 +0100
@@ -120,7 +120,7 @@
 ########################################
 # Build common code library            #
 ########################################
-include_directories(${CMAKE_SOURCE_DIR} log4cpp)
+include_directories(${CMAKE_SOURCE_DIR})
 
 add_definitions(-DADOBE_STD_SERIALIZATION)
 
@@ -196,30 +196,13 @@
     util/binreloc.c
     util/Directories.cpp
     util/XMLDoc.cpp
-    log4cpp/src/Appender.cpp
-    log4cpp/src/AppenderSkeleton.cpp
-    log4cpp/src/BasicLayout.cpp
-    log4cpp/src/Category.cpp
-    log4cpp/src/CategoryStream.cpp
-    log4cpp/src/Configurator.cpp
-    log4cpp/src/FactoryParams.cpp
-    log4cpp/src/FileAppender.cpp
-    log4cpp/src/HierarchyMaintainer.cpp
-    log4cpp/src/LayoutAppender.cpp
-    log4cpp/src/Localtime.cpp
-    log4cpp/src/LoggingEvent.cpp
-    log4cpp/src/NDC.cpp
-    log4cpp/src/PatternLayout.cpp
-    log4cpp/src/Priority.cpp
-    log4cpp/src/StringUtil.cpp
-    log4cpp/src/TimeStamp.cpp
     ${CMAKE_BINARY_DIR}/util/Version.cpp
 )
 
 if (WIN32)
     list(APPEND THIS_LIB_SOURCES log4cpp/src/MSThreads.cpp)
 else ()
-    list(APPEND THIS_LIB_SOURCES log4cpp/src/PThreads.cpp)
+    list(APPEND THIS_LIB_SOURCES)
 endif ()
 
 library_all_variants(core)
