--- doc/Makefile~	2012-03-14 21:03:58.000000000 +0100
+++ doc/Makefile	2013-06-17 16:44:52.195283030 +0200
@@ -10,7 +10,7 @@
 	pdflatex rtf2latexDoc
 
 web/manpage.html: header.html footer.html rtf2latex2e.1
-	man ./rtf2latex2e.1 | man2html --pgsize=100000 --bare > mid.html
+	man2html ./rtf2latex2e.1 > mid.html
 	# remove first four spaces from each lines
 	perl -pi.bak -e 's/^ {2}//' mid.html
 	cat header.html mid.html footer.html > man.html
