--- doc/Makefile~	2013-06-17 16:52:02.000000000 +0200
+++ doc/Makefile	2013-06-17 16:53:41.267276740 +0200
@@ -21,6 +21,8 @@
 	pdflatex rtf2latexDoc
 	pdflatex rtf2latexDoc
 	hevea rtf2latexDoc.tex
+	# Twice, hevea says cross references might have changed
+	hevea rtf2latexDoc.tex
 	tail -n +50 rtf2latexDoc.html > temp1.html
 	sed -n -e :a -e '1,7!{P;N;D;};N;ba' temp1.html > temp2.html
 	cat header.html temp2.html footer.html > temp3.html
