--- a/CPP/7zip/7zip_gcc.mak	2023-07-12 15:09:43.264812100 +0200
+++ b/CPP/7zip/7zip_gcc.mak	2023-07-12 15:11:34.326817937 +0200
@@ -142,7 +142,7 @@
 
 # LOCAL_LIBS=-lpthread
 # LOCAL_LIBS_DLL=$(LOCAL_LIBS) -ldl
-LIB2 = -lpthread -ldl
+LIB2 = -lpthread -ldl -z noexecstack
 
 
 endif
@@ -208,7 +208,7 @@
 # -Wl,--print-gc-sections
 
 ifneq ($(CC), $(CROSS_COMPILE)clang)
-LFLAGS_STRIP = -s
+LFLAGS_STRIP =
 endif
 
 LFLAGS_ALL = $(LFLAGS_STRIP) $(MY_ARCH_2) $(LDFLAGS) $(FLAGS_FLTO) $(LD_arch) $(OBJS) $(MY_LIBS) $(LIB2)
--- a/C/7zip_gcc_c.mak	2023-07-12 15:41:03.355910913 +0200
+++ b/C/7zip_gcc_c.mak	2023-07-12 15:41:35.353912595 +0200
@@ -121,7 +121,7 @@
 
 # LOCAL_LIBS=-lpthread
 # LOCAL_LIBS_DLL=$(LOCAL_LIBS) -ldl
-LIB2 = -lpthread -ldl
+LIB2 = -lpthread -ldl -z noexecstack
 
 DEL_OBJ_EXE = -$(RM) $(PROGPATH) $(PROGPATH_STATIC) $(OBJS)
 
