--- a/cegui/include/CEGUI/RendererModules/Ogre/Renderer.h	2018-07-30 19:44:32.102575772 +0200
+++ b/cegui/include/CEGUI/RendererModules/Ogre/Renderer.h	2018-07-30 19:45:39.480573461 +0200
@@ -31,6 +31,16 @@
 #include "../../Size.h"
 #include "../../Vector.h"
 #include "CEGUI/Config.h"
+#include <OGRE/OgreBuildSettings.h>
+
+// This is only set in Ogre-1.10
+#ifndef OGRE_USE_STD11
+		#define OGRE_USE_STD11 0
+#endif // Check OGRE_USE_STD11
+
+#if OGRE_USE_STD11
+	#include <memory>
+#endif
 
 #include <vector>
 
@@ -57,7 +67,11 @@
 #if (CEGUI_OGRE_VERSION < ((1 << 16) | (9 << 8) | 0))
 class TexturePtr;
 #else
-template<typename T> class SharedPtr;
+#if OGRE_USE_STD11
+    template<typename T> using SharedPtr = std::shared_ptr<T>;
+#else
+    template<typename T> class SharedPtr;
+#endif
 class Texture;
 typedef SharedPtr<Texture> TexturePtr;
 #endif
