--- plugins/propclass/mechanics/mechanics.cpp~	2012-07-02 12:55:27.000000000 +0200
+++ plugins/propclass/mechanics/mechanics.cpp	2012-07-13 16:48:39.875066116 +0200
@@ -89,8 +89,6 @@
     AddAction (action_setsystem, "SetSystem");
     AddAction (action_setgravity, "SetGravity");
     AddAction (action_quickstep, "QuickStep");
-    AddAction (action_enablestepfast, "EnableStepFast");
-    AddAction (action_disablestepfast, "DisableStepFast");
     AddAction (action_setsteptime, "SetStepTime");
     AddAction (action_setsimulationspeed, "SetSimulationSpeed");
   }
@@ -246,29 +244,11 @@
     dynsystem->QueryObject ()->SetName ("cel.mechanics.system");
     dynsystem->SetGravity (csVector3 (0, -9.8f, 0));
 
-    EnableStepFast ();
+    EnableQuickStep ();
   }
   return dynsystem;
 }
 
-void celPcMechanicsSystem::DisableStepFast ()
-{
-  GetDynamicSystem ();
-  if (!dynsystem) return;
-  csRef<iODEDynamicSystemState> osys= scfQueryInterface<iODEDynamicSystemState> (dynsystem);
-  if (osys)
-    osys->EnableStepFast (0);
-}
-
-void celPcMechanicsSystem::EnableStepFast ()
-{
-  GetDynamicSystem ();
-  if (!dynsystem) return;
-  csRef<iODEDynamicSystemState> osys= scfQueryInterface<iODEDynamicSystemState> (dynsystem);
-  if (osys)
-    osys->EnableStepFast (1);
-}
-
 void celPcMechanicsSystem::EnableQuickStep ()
 {
   GetDynamicSystem ();
@@ -323,14 +303,11 @@
         }
         return true;
       }
+    case action_enablestepfast:
     case action_quickstep:
       EnableQuickStep ();
       return true;
-    case action_enablestepfast:
-      EnableStepFast ();
-      return true;
     case action_disablestepfast:
-      DisableStepFast ();
       return true;
     case action_setsteptime:
       {
