--- scripts/blcelc.py~	2012-07-02 12:55:28.000000000 +0200
+++ scripts/blcelc.py	2012-07-13 16:42:50.255047828 +0200
@@ -1256,8 +1256,6 @@
     def RemoveBodyFromGroup(*args): return _blcelc.iPcMechanicsSystem_RemoveBodyFromGroup(*args)
     def CreateJoint(*args): return _blcelc.iPcMechanicsSystem_CreateJoint(*args)
     def RemoveJoint(*args): return _blcelc.iPcMechanicsSystem_RemoveJoint(*args)
-    def DisableStepFast(*args): return _blcelc.iPcMechanicsSystem_DisableStepFast(*args)
-    def EnableStepFast(*args): return _blcelc.iPcMechanicsSystem_EnableStepFast(*args)
     def EnableQuickStep(*args): return _blcelc.iPcMechanicsSystem_EnableQuickStep(*args)
     DynamicSystem = _swig_property(_blcelc.iPcMechanicsSystem_GetDynamicSystem, fix_args(_blcelc.iPcMechanicsSystem_SetDynamicSystem), None,
                     "iPcMechanicsSystem.DynamicSystem -> type\n\nThis is equivalent to calling the C++ cs methods:\n\tget: iPcMechanicsSystem::GetDynamicSystem()\n\tset: void iPcMechanicsSystem::SetDynamicSystem(...)")
