--- a/include/host_config.h	2014-08-08 19:37:24.000000000 +0200
+++ b/include/host_config.h	2014-10-30 14:14:15.622996929 +0100
@@ -77,11 +77,11 @@
 
 #if defined(__GNUC__)
 
-#if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 8)
+#if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 9)
 
-#error -- unsupported GNU version! gcc 4.9 and up are not supported!
+#error -- unsupported GNU version! gcc 4.10 and up are not supported!
 
-#endif /* __GNUC__> 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 8) */
+#endif /* __GNUC__> 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 9) */
 
 #endif /* __GNUC__ */
 
