--- a/include/host_config.h	2015-03-06 09:43:05.000000000 +0100
+++ b/include/host_config.h	2015-09-03 18:30:08.733054542 +0200
@@ -100,11 +100,11 @@
 
 #if defined(__GNUC__)
 
-#if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 9)
+#if __GNUC__ > 5 || (__GNUC__ == 5 && __GNUC_MINOR__ > 2)
 
-#error -- unsupported GNU version! gcc 4.10 and up are not supported!
+#error -- unsupported GNU version! gcc 5.3 and up are not supported!
 
-#endif /* __GNUC__> 4 || (__GNUC__ == 4 && __GNUC_MINOR__ > 9) */
+#endif /* __GNUC__> 5 || (__GNUC__ == 5 && __GNUC_MINOR__ > 2) */
 
 #if defined(__APPLE__) && defined(__MACH__) && !defined(__clang__)
 #error -- clang and clang++ are the only supported host compilers on Mac OS X!
