--- src/opengl/qgl.cpp  2011-12-08 06:06:03.000000000 +0100
+++ src/opengl/qgl.cpp  2012-02-10 09:34:57.428827696 +0100
@@ -5621,7 +5621,7 @@
         return QLatin1String("/System/Library/Frameworks/OpenGL.framework/Versions/A/Libraries/libGL.dylib");
 #else
 # if defined(QT_OPENGL_ES_1)
-        return QLatin1String("GLES_CM");
+        return QLatin1String("GLESv1_CM");
 # elif defined(QT_OPENGL_ES_2)
         return QLatin1String("GLESv2");
 # else
--- mkspecs/common/linux.conf	2011-12-08 06:06:02.000000000 +0100
+++ mkspecs/common/linux.conf	2012-02-10 09:41:58.035516361 +0100
@@ -30,7 +30,7 @@
 QMAKE_LIBS_EGL        = -lEGL
 QMAKE_LIBS_OPENGL     = -lGL
 QMAKE_LIBS_OPENGL_QT  = -lGL
-QMAKE_LIBS_OPENGL_ES1 = -lGLES_CM
+QMAKE_LIBS_OPENGL_ES1 = -lGLESv1_CM
 QMAKE_LIBS_OPENGL_ES2 = -lGLESv2
 QMAKE_LIBS_OPENVG     = -lOpenVG
 QMAKE_LIBS_THREAD     = -lpthread
