--- a/v4l2loopback.c	2020-04-27 11:12:31.550824848 +0200
+++ b/v4l2loopback.c	2020-04-27 11:31:25.550780536 +0200
@@ -16,7 +16,16 @@
 #include <linux/version.h>
 #include <linux/vmalloc.h>
 #include <linux/mm.h>
-#include <linux/time.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 6, 0)
+# include <uapi/linux/time.h>
+# include <uapi/linux/module.h>
+# include <uapi/linux/videodev2.h>
+# include <uapi/linux/v4l2-common.h>
+# define timeval __kernel_v4l2_timeval
+  typedef __kernel_old_time_t time_t;
+#else
+# include <linux/time.h>
+#endif
 #include <linux/module.h>
 #include <linux/videodev2.h>
 #include <linux/sched.h>
