--- GG/GG/adobe/cmath.hpp	2012-06-15 20:39:26.000000000 +0200
+++ GG/GG/adobe/cmath.hpp	2012-06-15 20:49:54.583897546 +0200
@@ -35,16 +35,16 @@
 // Guessing at gcc 3 support
 #if  (__GNUC__ == 3) && (__GNUC_MINOR__ > 2)
 
-#define ADOBE_HAS_CPP_CMATH 
+#define ADOBE_HAS_CPP_CMATH
 
-#elif ((__GNUC__ == 4) && (__GNUC_MINOR__ <= 6))
+#elif ((__GNUC__ == 4) && (__GNUC_MINOR__ <= 7))
 /*
     The currently supported version of GNUC has C99 extensions in math.h. But no TR1 extensions.
 */
 #define ADOBE_HAS_C99_MATH_H
  
 #else
-#error "Unknown GCC compiler configuration for cmath (last known version is 4.0.1)."
+#error "Unknown GCC compiler configuration for cmath (last known version is 4.7.1)."
 #endif
 
 #elif defined(_MSC_VER)
