--- a/megasync-plugin.h	2017-12-29 09:49:52.180449208 +0100
+++ b/megasync-plugin.h	2017-12-29 09:49:21.240448279 +0100
@@ -5,6 +5,7 @@
 #ifndef WITH_KF5
 #include <kabstractfileitemactionplugin.h>
 #else
+#include <KPluginFactory>
 #include <KIOWidgets/kabstractfileitemactionplugin.h>
 #endif
 #include <QLocalSocket>
@@ -34,4 +35,6 @@
     QString getString(int type, int numFiles,int numFolders);
 };
 
+K_PLUGIN_FACTORY_DECLARATION(MEGASyncPluginFactory)
+
 #endif
--- a/megasync-plugin.cpp	2017-12-29 09:49:47.056449054 +0100
+++ b/megasync-plugin.cpp	2017-12-29 09:49:21.240448279 +0100
@@ -11,7 +11,6 @@
 #include <KDE/KPluginFactory>
 #include <KDE/KPluginLoader>
 #else
-#include <KPluginFactory>
 #include <KPluginLoader>
 #include <KIOWidgets/kabstractfileitemactionplugin.h>
 #include <QtNetwork/QLocalSocket>
@@ -28,9 +27,6 @@
 
 #include "megasync-plugin.h"
 
-K_PLUGIN_FACTORY(MEGASyncPluginFactory, registerPlugin<MEGASyncPlugin>();)
-K_EXPORT_PLUGIN(MEGASyncPluginFactory("megasync-plugin"))
-
 
 typedef enum {
     STRING_UPLOAD = 0,
@@ -80,7 +76,7 @@
 {
     Q_UNUSED(parentWidget);
     QList<QAction*> actions;
-    int state;
+    int state = 0;
 
     int syncedFiles, syncedFolders, unsyncedFiles, unsyncedFolders;
     syncedFiles = syncedFolders = unsyncedFiles = unsyncedFolders = 0;
@@ -285,4 +281,7 @@
     return reply;
 }
 
+K_PLUGIN_FACTORY_DEFINITION(MEGASyncPluginFactory, registerPlugin<MEGASyncPlugin>();)
+K_EXPORT_PLUGIN(MEGASyncPluginFactory("megasync-plugin"))
+
 #include "megasync-plugin.moc"
