--- a/meson/find_source.sh	2019-09-25 17:49:05.413584797 +0200
+++ b/meson/find_source.sh	2019-09-25 17:53:55.816585713 +0200
@@ -2,15 +2,14 @@
 
 # Wrappers are provided from the call
 for x in $@ ; do
-        echo "$x" | sed -e "s,$MESON_SOURCE_ROOT,..,g"
+        echo "$x"
 done
 
+
 # Headers must be found and go first
-find "$MESON_SOURCE_ROOT"/src/ -type f -name '*.h' | \
-        sort -u | \
-        sed -e "s,$MESON_SOURCE_ROOT,..,g"
+find "$MESON_SOURCE_ROOT"/src/ -type f -name '*.h' | sort -u
+
 
 # Followed by the source files
-find "$MESON_SOURCE_ROOT"/src/ -type f -name '*.cpp' | \
-        sort -u | \
-        sed -e "s,$MESON_SOURCE_ROOT,..,g"
+find "$MESON_SOURCE_ROOT"/src/ -type f -name '*.cpp' | sort -u
+
