--- a/CMakeLists.txt	2025-07-08 10:15:02.617920242 +0200
+++ b/CMakeLists.txt	2025-07-08 10:15:13.626568677 +0200
@@ -162,10 +162,6 @@
     message(STATUS "Enabled MIDI support (Haiku)")
     add_subdirectory(src/midi)
 else()
-    # Workaround for SDL bug #3295, which occurs in SDL2 <2.0.5
-    # https://bugzilla.libsdl.org/show_bug.cgi?id=3295
-    cmake_policy(SET CMP0004 OLD)
-
     find_package(SDL2 REQUIRED)
 endif()
 
