--- a/src/libaccountsservice/act-user-manager.c	2016-01-18 19:07:01.000000000 +0100
+++ b/src/libaccountsservice/act-user-manager.c	2016-11-01 17:22:28.449045864 +0100
@@ -42,7 +42,7 @@
 #include <gio/gunixinputstream.h>
 
 #ifdef WITH_SYSTEMD
-#include <systemd/sd-login.h>
+#include <elogind/sd-login.h>
 
 /* check if logind is running */
 #define LOGIND_RUNNING() (access("/run/systemd/seats/", F_OK) >= 0)
