--- a/wscript	2015-09-30 15:03:26.862409759 +0200
+++ b/wscript	2015-09-30 15:03:45.054410305 +0200
@@ -91,6 +91,7 @@
 	if Options.options.graphics:
 		conf.check_ogre()
 		conf.check_ogre_plugindir()
+		conf.check_ogre_overlay()
 		conf.check_ogre_terrain()
 		conf.check_ois()
 		conf.check_xlib()
@@ -509,6 +510,25 @@
 	conf.end_msg(False)
 
 @conf
+def check_ogre_overlay(conf):
+	conf.start_msg('Checking for package OGRE-Overlay')
+	conf.env.stash()
+	if conf.check_cfg(package='OGRE-Overlay', atleast_version='1.7.0', args='--cflags --libs', mandatory=False, uselib_store='OGRE'):
+		conf.end_msg('pkg-config OGRE-Overlay')
+		return
+	conf.env.revert()
+	conf.env.stash()
+	if conf.check_cxx(lib='OgreOverlay', mandatory=False, uselib='TEST', uselib_store='OGRE') and\
+	   conf.check_cxx(header_name='Overlay/OgreOverlay.h', mandatory=False, uselib='TEST', uselib_store='OGRE', fragment='''
+			#include <stdio.h>
+			int main() { printf(""); return 0; }'''):
+		conf.end_msg('library OgreOverlay')
+		return
+	conf.env.revert()
+	conf.end_msg(False)
+	conf.fatal('Could not find OGRE-Overlay')
+
+@conf
 def check_ogre_terrain(conf):
 	conf.start_msg('Checking for package OGRE-Terrain')
 	conf.env.stash()
