--- a/src/lipsofsuna/render/overlay/render-overlay.cpp	2015-09-30 14:42:28.730371987 +0200
+++ b/src/lipsofsuna/render/overlay/render-overlay.cpp	2015-09-30 14:53:29.152391815 +0200
@@ -537,7 +537,8 @@
 	}
 
 	/* Create a new material. */
-	material = self->render->material_manager->load (matname, group);
+	self->render->material_manager->load (matname, group);
+	material = self->render->material_manager->getByName (matname, group);
 	lisys_assert (!material.isNull ());
 	material->setSceneBlending (Ogre::SBT_TRANSPARENT_ALPHA);
 
