--- a/src/lipsofsuna/extension/particle/particle-model-system.cpp	2017-05-17 11:46:02.912635866 +0200
+++ b/src/lipsofsuna/extension/particle/particle-model-system.cpp	2017-05-17 11:48:36.374633118 +0200
@@ -85,20 +85,20 @@
 			particle->mOwnDimensions = false;
 			particle->mWidth = 2.0f * system->particle_size;
 			particle->mHeight = 2.0f * system->particle_size;
-			particle->rotation = 0.0f;
-			particle->position = Ogre::Vector3 (pos.x, -pos.y, pos.z);
-			particle->direction = Ogre::Vector3 (0.0f, 0.0f, 0.0f);
-			particle->colour = Ogre::ColourValue (color[0], color[1], color[2], color[3]);
-			particle->timeToLive = 1000.0f;
-			particle->totalTimeToLive = 1000.0f;
-			particle->rotationSpeed = 0.0f;
-			particle->particleType = Ogre::Particle::Visual;
+			particle->mRotation = 0.0f;
+			particle->mPosition = Ogre::Vector3 (pos.x, -pos.y, pos.z);
+			particle->mDirection = Ogre::Vector3 (0.0f, 0.0f, 0.0f);
+			particle->mColour = Ogre::ColourValue (color[0], color[1], color[2], color[3]);
+			particle->mTimeToLive = 1000.0f;
+			particle->mTotalTimeToLive = 1000.0f;
+			particle->mRotationSpeed = 0.0f;
+			particle->mParticleType = Ogre::Particle::Visual;
 		}
 		else if (created[i])
 		{
 			Ogre::Particle* particle = created[i];
-			particle->timeToLive = 0.0f;
-			particle->totalTimeToLive = 0.0f;
+			particle->mTimeToLive = 0.0f;
+			particle->mTotalTimeToLive = 0.0f;
 			created[i] = NULL;
 		}
 	}
