--- a/src/lipsofsuna/render/internal/render-terrain-material-generator.hpp	2017-05-17 11:35:06.934647613 +0200
+++ b/src/lipsofsuna/render/internal/render-terrain-material-generator.hpp	2017-05-17 11:35:27.350647248 +0200
@@ -38,10 +38,11 @@
 #define __RENDER_INTERNAL_TERRAIN_MATERIAL_GENERATOR_HPP__
 
 #include "lipsofsuna/system.h"
-#include "OgreTerrainPrerequisites.h"
-#include "OgreTerrainMaterialGenerator.h"
-#include "OgreGpuProgramParams.h"
-#include "OgreShadowCameraSetupPSSM.h"
+#include <Ogre.h>
+#include <OgreTerrainPrerequisites.h>
+#include <OgreTerrainMaterialGenerator.h>
+#include <OgreGpuProgramParams.h>
+#include <OgreShadowCameraSetupPSSM.h>
 
 using namespace Ogre;
 
