--- a/src/lipsofsuna/render/overlay/render-overlay.hpp	2017-05-17 11:39:30.575642892 +0200
+++ b/src/lipsofsuna/render/overlay/render-overlay.hpp	2017-05-17 11:42:30.935639662 +0200
@@ -21,6 +21,7 @@
 #include "lipsofsuna/math.h"
 #include "lipsofsuna/render/internal/render.hpp"
 #if OGRE_VERSION_MAJOR >= 1 && OGRE_VERSION_MINOR >= 9
+#include <Overlay/OgreOverlay.h>
 #include <Overlay/OgreOverlaySystem.h>
 #include <Overlay/OgreOverlayManager.h>
 #include <Overlay/OgreOverlayElementFactory.h>
--- a/src/lipsofsuna/extension/physics-debug/render-physics-debug.hpp	2017-05-17 11:51:02.807630495 +0200
+++ b/src/lipsofsuna/extension/physics-debug/render-physics-debug.hpp	2017-05-17 11:52:49.630628582 +0200
@@ -10,7 +10,12 @@
 #include <btBulletCollisionCommon.h>
 #include <OgreFrameListener.h>
 #include <OgreSceneManager.h>
+#include <OgreSceneNode.h>
 #include <OgreManualObject.h>
+#include <OgreTechnique.h>
+#include <OgreTimer.h>
+#include <OgreLogManager.h>
+#include <OgreMaterialManager.h>
 
 class OgreDebugDrawer: public btIDebugDraw, public Ogre::FrameListener{
 public:
